src/lib/support/logging/CHIPLogging.cpp

```diff
@@ -199,6 +199,115 @@ DLL_EXPORT void SetLogFilter(uint8_t category)
 }
 #endif // CHIP_LOG_FILTERING
 
+/**
+ * This static method outputs a line of the memory dump.
+ *
+ * @param[in]  module    The LogModule generating the log message.
+ * @param[in]  category  The LogCategory for filtering log level.
+ * @param[in]  aBuf      A pointer to the buffer.
+ * @param[in]  aLength   Number of bytes in the buffer.
+ *
+ */
+static void DumpLine(uint8_t module, uint8_t category, const void * aBuf, const size_t aLength)
+{
+    char buf[80];
+    char * cur = buf;
+
+    snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), "|");
+    cur += strlen(cur);
+
+    for (size_t i = 0; i < 16; i++)
+    {
+        if (i < aLength)
+        {
+            snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), " %02X", ((uint8_t *) (aBuf))[i]);
+            cur += strlen(cur);
+        }
+        else
+        {
+            snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), " ..");
+            cur += strlen(cur);
+        }
+
+        if (!((i + 1) % 8))
+        {
+            snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), " |");
+            cur += strlen(cur);
+        }
+    }
+
+    snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), " ");
+    cur += strlen(cur);
+
+    for (size_t i = 0; i < 16; i++)
+    {
+        char c = 0x7f & ((char *) (aBuf))[i];
+
+        if (i < aLength && isprint(c))
+        {
+            snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), "%c", c);
+            cur += strlen(cur);
+        }
+        else
+        {
+            snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), ".");
+            cur += strlen(cur);
+        }
+    }
+
+    Log(module, category, "%s", buf);
+}
+
+/**
+ * This method dumps bytes to the log in a human-readable fashion.
+ *
+ * @param[in]  module    The LogModule generating the log message.
+ * @param[in]  category  The LogCategory for filtering log level.
+ * @param[in]  msg       A pointer to a NULL-terminated string that is printed before the bytes.
+ * @param[in]  aBuf      A pointer to the buffer.
+ * @param[in]  aLength   Number of bytes to print.
+ *
+ */
+void Dump(uint8_t module, uint8_t category, const char * msg, const void * aBuf, const unsigned aLength)
```

Comment:
If I were using this interface, I would personally prefer to retain the ability to pass printf format strings.  That implies you would have interfaces like these:

```
void DumpV(uint8_t module, uint8_t category, const void * buf, const size_t length, const char * msg, va_list args);
void Dump(uint8_t module, uint8_t category, const void * buf, const size_t length, const char * msg, ...) ENFORCE_FORMAT(5, 6);
```